src/xdocs/config_design.xml

```diff
@@ -1508,39 +1508,37 @@ class Test {
         </source>
         <p>Example with default values:</p>
         <source>
-public class MyClass {
- private int myPrivateField1;
+class Example1 {
+  private int myPrivateField1;
 
- int field1;              // violation, must have a visibility modifier
+  int field1; // violation 'must be private'
 
- protected String field2; // violation, protected visibility is not allowed
+  protected String field2; // violation 'must be private'
 
- public int field3 = 42;  // violation, not static final, immutable,
-                          // nor matching the pattern
+  public int field3 = 42; // violation 'must be private'
 
- public long serialVersionUID = 1L;
+  public long serialVersionUID = 1L;
 
- public static final int field4 = 42;
+  public static final int field4 = 42;
 
- public final int field5 = 42;          // violation, public immutable fields are not allowed
- public final java.lang.String notes;   // violation, public immutable fields are not allowed
+  public final int field5 = 42; // violation 'must be private'
```

Comment:
Same argument as the other PR. I can see why we are requiring it to be just the message, but we are losing some understanding why it must be that which is not explained in the violation message.

Reply:
@stoyanK7 , please try to make it
`// violation, public immutable fields are not allowed 'mus be private'`

It is ok to extend our code a bit in abstract test to allow this

Reply:
@rnveach @romani Is adding the entire message only on the first violation okay? Placing the entire violation message on each `// violation` here would make the test be
```
code
// violation above
code
// violation above
```

Reply:
I didn't not mean whole message of violation. I meant explanation wording for violation.

But yeah we need to follow line length, we have to follow, and place comment on line above or below.

Reply:
> // violation, public immutable fields are not allowed 'must be private'

This is not allowed. Inline violation comments are strict `// violation '<violation message>'` 

https://github.com/checkstyle/checkstyle/blob/3f25c37d731544f47d13020b47bfbec2a1bf57b6/src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java#L48-L49

We already had a discussion to keep violation messages until we see a case where they don't do a good job. Please see https://github.com/checkstyle/checkstyle/pull/13112#discussion_r1210788609

Reply:
Please change `\\s*` to `.*` to let it be supported format

Reply:
Wouldn't it be also good to limit this violation pattern to examples only?

Reply:
I think we can benefit from it even in Inputs.
We can start use it only Examples if required, and may be somebody will start using it in inputs later on.

Reply:
Somewhat done. The pattern is having issues with lots of tests. The line `final String violationMessage = violationMatcher.group(1);` is `null`

Reply:
Pitest is massively failing

Reply:
And general tests

Reply:
@romani Done. Maybe the code can be further extracted into methods to avoid some duplication. Let me know if that should be done.

Reply:
It is ok for now